### PR TITLE
Added code for flattenDeepArray

### LINKED ISSUE #195: Fix flattenDeepArray method
There is currently a method called `flattenDeepArray` that lives in `/lib/flattenDeepArray.js`.
It is incomplete and needs to be fixed!

### Acceptance Criteria:
1. Running `yarn test <methodName>` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by hvarday**
- [] Running `yarn lint` does not trigger any linter errors
- [] The test of the method you have fixed is passing
- [] You have written a new skeleton method for someone else to work on!
- [] You have written tests to accompany your skeleton method

Closes #195

### CODE REVIEW

File: test/flattenNestedObject.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const parameterInput1 = { a : 1, b : "banana", c : { d : [1, 2, 3] } },
+      output1 = { "a" : 1, "b" : "banana", "c.d" : [1, 2, 3] },
+      parameterInput2 = { "a" : 1, "b" : { "c" : { "d" : { "e" : 1, "f" : 2 } } } },
+      output2 = {"a": 1, "b.c.d.e": 1, "b.c.d.f": 2};
+
+    expect(flattenNestedObject(parameterInput1)).toEqual(output1);
+    expect(flattenNestedObject(parameterInput2)).toEqual(output2);
+  });
+
+  it("should return the flatten object of the passed nested object parameter with default delimiter(.)", () => {
```
Comment (ktilcu) [inline test/flattenNestedObject.test.js]: I think you meant to adjust this statement to be "-" in stead of default delimiter.

Comment (hvarday) [reply test/flattenNestedObject.test.js]: yes, that was supposed to be "-" delimiter. I'll fix it and push a commit.

Comment (hvarday) [reply test/flattenNestedObject.test.js]: Fixed the description.

Comment (ktilcu) [reply test/flattenNestedObject.test.js]: 👍

**Comment by ktilcu** [comment]
Can you rebase so we can merge this?

**Comment by BlakeGuilloud** [comment]
Awesome @hvarday ! Thanks for your work on this. Going to go ahead and merge this puppy in. We are trying to encourage our contributors to open a new Issue for whatever skeleton method they introduce, so if you have some time to do that, it helps us out a ton! Thanks!!

**Pull request merged**